lib/iris/_data_manager.py

```diff
@@ -185,7 +199,15 @@ def _assert_axioms(self):
                 'real data and realised {!r}.')
         assert state, emsg.format(self._realised_dtype)
 
-    def _deepcopy(self, memo, data=None, realised_dtype='none'):
+        state = not (self.has_lazy_data() and
+                     self._lazy_array.dtype.kind != 'f' and
+                     self._realised_dtype is not None)
```

Comment:
Are you sure this is right? It seems so me I should be allowed to have lazy integer data and not pass a realised_dtype,